dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/eutaxonomynonfinancials/model/validator/ReferencedReportsListValidator.kt

```diff
@@ -0,0 +1,60 @@
+// <--WARNING--> THIS FILE IS AUTO-GENERATED BY THE FRAMEWORK-TOOLBOX AND WILL BE OVERWRITTEN
+package org.dataland.datalandbackend.frameworks.eutaxonomynonfinancials.model.validator
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import org.dataland.datalandbackend.frameworks.eutaxonomynonfinancials.model.EutaxonomyNonFinancialsData
+import org.dataland.datalandbackend.model.documents.CompanyReport
+import kotlin.reflect.KClass
+
+/**
+* Annotation for the validation of referenced reports list for eutaxonomy-non-financials
+*/
+
+@Constraint(validatedBy = [ReferencedReportsListConstraintValidator::class])
+@Target(AnnotationTarget.FIELD, AnnotationTarget.CLASS)
+annotation class ReferencedReportsListValidator(
+    val message: String = "The list of referenced reports is not complete.",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<out Payload>> = [],
+)
+
+/**
+* Class holding the validation logic for referenced reports list. It checks if the referenced reports list is complete
+*/
+class ReferencedReportsListConstraintValidator :
+    ConstraintValidator<ReferencedReportsListValidator, EutaxonomyNonFinancialsData> {
+
+    override fun isValid(dataset: EutaxonomyNonFinancialsData?, context: ConstraintValidatorContext?): Boolean {
+        if (dataset == null) {
+            return false
+        }
+        val referencedReportsMap = dataset.general?.referencedReports
+        val referencedReportsFileReference = getFileReferencesFromReports(referencedReportsMap)
+
+        val extendedDocumentsFileReferences = getExtendedDocumentReferences(dataset).filterNotNull()
+        return extendedDocumentsFileReferences.all {
+            referencedReportsFileReference.contains(it)
+        }
+    }
+
+    private fun getFileReferencesFromReports(map: Map<String, CompanyReport>?): List<String> {
+        if (map == null) return emptyList()
+        val referencedReportsList = mutableListOf<String>()
+        for (entry in map.entries) {
+            referencedReportsList.add(entry.value.fileReference)
+        }
+        return referencedReportsList
+    }
+
+    @Suppress("MaxLineLength", "LongMethod")
+    private fun getExtendedDocumentReferences(dataset: EutaxonomyNonFinancialsData): List<String?> {
+        return listOf(
```

Comment:
in Eu-Taxo non financial there is also an extended reference under General

Reply:
added the missing extended referenced (its a custom component that claim to have no document support)